rust/lance/src/dataset/write/commit.rs

```diff
@@ -190,22 +190,23 @@ impl<'a> CommitBuilder<'a> {
             }
         };
 
+        let session = self
+            .session
+            .or_else(|| self.dest.dataset().map(|ds| ds.session.clone()))
+            .unwrap_or_default();
+
         let dest = match &self.dest {
             WriteDestination::Dataset(dataset) => WriteDestination::Dataset(dataset.clone()),
             WriteDestination::Uri(uri) => {
                 // Check if it already exists.
-                let mut builder = DatasetBuilder::from_uri(uri).with_read_params(ReadParams {
-                    store_options: self.store_params.clone(),
-                    commit_handler: self.commit_handler.clone(),
-                    object_store_registry: self.object_store_registry.clone(),
-                    ..Default::default()
-                });
-
-                // If read_version is zero, then it might not have originally been
-                // passed. We can assume the latest version.
-                if transaction.read_version > 0 {
-                    builder = builder.with_version(transaction.read_version)
```

Comment:
Do we not need to pass in the read_version anymore?

Reply:
I think, we only *need* to load the `read_version` if it's a detached commit. I've fixed this in the latest commits.